### PR TITLE
Names references, join, multiple sources

### PR DESCRIPTION
**Pull request by dkrasner**
### Main Points ###

Our grammar and semantics now allow for named references, which you can see [here](https://github.com/dkrasner/worksheet/compare/daniel-named-refs?expand=1#diff-8fcd69e406d7295a9fcc0c26dfd9da643fcf32752196a53e2e05cb7fd322609aR12). Although this is not implemented in any of the function-instructions at the moment it would be relatively straightforward to do so. I imagine in the future we'll want to allow for reasonably complex operations which will require keeping track of references. For example, a multiple source selection which joins 2 sources on a string, joins another 2 sources on a string, and then joins the result on yet another string and then writes that into a target subframe. All of this depends on how we craft the UX of the actual "programming" in the environment, which is still not clear to me, but if I am wrong and we never use named references/variables then we can easily remove it from the grammar and semantics. 

Our default interpreter now uses the sheet `.apply()`, `.getDataSubFrame()` and related API's and doesn't have to worry about manipulating coordinates, cells and data therein which makes things much cleaner in my view. [Here](https://github.com/dkrasner/worksheet/compare/daniel-named-refs?expand=1#diff-847e9549fa75505e8b2286b4bc0a789fc6e9ab285c9fd9c7848320507c39d6fbR100) is an example. 

There is a new [join(string)](https://github.com/dkrasner/worksheet/compare/daniel-named-refs?expand=1#diff-847e9549fa75505e8b2286b4bc0a789fc6e9ab285c9fd9c7848320507c39d6fbR100) function which takes takes two sources and joins their respective elements over the provided string. As mentioned above nothing happens with named references at the moment - `join()` assumes that you join in order that the source references appear. But in the future we could imagine something more flexible like `join(aRef, anotherRef, ",")`....
 
Tests have been written for everything above. 

Finally there is a [mapper.html](https://github.com/dkrasner/worksheet/compare/daniel-named-refs?expand=1#diff-67d474309a1f03fcae378b9ceeecd70c65956c9986e77ebf5ace50b0ceed9ac7) in `/examples` so you don't have to create and populate worksheets to try things out.

### CODE REVIEW

File: js/Worksheet.js [hunk 1]
```diff
@@ -707,6 +716,8 @@ class Worksheet extends HTMLElement {
 
     /* I set the sheet.selection for a worksheet */
     select(id, coordinates){
+        console.log(id)
```
Comment (darth-cheney) [inline js/Worksheet.js:719]: We can prob delete this and the line below it

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "./dist/worksheet.bundle.js",
   "type": "module",
   "scripts": {
-    "test": "./node_modules/mocha/bin/mocha.js --require ./tests/preload.js ./tests/*",
+    "test": "./node_modules/mocha/bin/mocha.js --require --timeout 3000 ./tests/preload.js ./tests/*",
```
Comment (darth-cheney) [inline package.json:8]: This will not work for running the tests. You need to put `--require` in front of the preload location

**Review by darth-cheney — CHANGES_REQUESTED** [review]
Aside from the small inline comments, looks good. Tests passing, and the example worksheet is nice. Will need to think about how to best use Refs as you say

File: package.json (hunk 2 above)
Comment (dkrasner) [reply package.json:8]: weird that it __does__ work

**Pull request merged**